engine/crates/engine-v2/src/sources/graphql/mod.rs

```diff
@@ -92,20 +133,65 @@ impl GraphqlPreparedExecutor {
                 subgraph_name: subgraph.name(),
                 timeout: subgraph.timeout(),
             },
-            |bytes| {
-                let response = subgraph_response.as_mut();
-                let status = GraphqlResponseSeed::new(
-                    response.next_seed(plan).ok_or("No object to update")?,
-                    RootGraphqlErrors {
-                        response,
-                        response_keys: plan.response_keys(),
-                    },
-                )
-                .deserialize(&mut serde_json::Deserializer::from_slice(&bytes))?;
-                Ok((status, subgraph_response))
+            GraphqlIngester {
+                ctx,
+                plan,
+                cache_key,
+                subgraph_response,
+                cache_ttl,
             },
         )
         .instrument(span)
         .await
     }
 }
+
+fn build_cache_key(json_body: &str) -> String {
+    let mut hasher = sha2::Sha256::new();
+    hasher.update(json_body);
+    let output = hasher.finalize();
+    output.encode_hex()
+}
+
+struct GraphqlIngester<'ctx, R: Runtime> {
+    ctx: ExecutionContext<'ctx, R>,
+    plan: PlanWalker<'ctx, (), ()>,
+    cache_key: Option<String>,
+    subgraph_response: SubgraphResponse,
+    cache_ttl: Duration,
+}
+
+impl<'ctx, R> ResponseIngester for GraphqlIngester<'ctx, R>
+where
+    R: Runtime,
+{
+    async fn ingest(
+        mut self,
+        bytes: Bytes,
+    ) -> Result<(GraphqlResponseStatus, SubgraphResponse), crate::execution::ExecutionError> {
+        if let Some(cache_key) = self.cache_key {
+            // We could probably put this call into the background at some point, but for
```

Comment:
hum indeed, there is a FutureUnordered executing background tasks here
https://github.com/grafbase/grafbase/blob/main/engine/crates/engine-v2/src/execution/coordinator.rs#L44

but no way to write in it, I guess we could have a MPSC channel with the sender inside the `ExecutionContext`.

Reply:
Yeah, was going to add a channel when/if we do this (not necessarily to that FutureUnordered, but maybe).  But out of scope for now

Reply:
yeah just wanted to mention that this existed for pretty much the same purpose, just at a different step of the execution. so could likely be re-used.